src/NumericalCalculations.jl

```diff
@@ -195,7 +195,7 @@ function _solveCCWNumeric!( ccwl::Union{CommonConvWrapper{F},
   
   # broadcast updates original view memory location
   ## using CalcFactor legacy path inside (::CalcFactor)
-  _hypoObj = (res, x) -> (target.=x; unrollHypo!(res) )
+  _hypoObj = (x) -> (target.=x; unrollHypo!())
```

Comment:
this has to remain `(res, x) ->` as standard for both Optim and NLsolve, and is used as such by both `approxConv` and `approxDeconv`

Reply:
https://github.com/JuliaRobotics/IncrementalInference.jl/blob/648aa8ca63eaadd69631b1532bc30cfd96efdb68/src/DeconvUtils.jl#L93

Reply:
https://github.com/JuliaRobotics/IncrementalInference.jl/blob/648aa8ca63eaadd69631b1532bc30cfd96efdb68/src/NumericalCalculations.jl#L46

Reply:
https://github.com/JuliaRobotics/IncrementalInference.jl/blob/648aa8ca63eaadd69631b1532bc30cfd96efdb68/src/NumericalCalculations.jl#L63

Reply:
I don't completely follow, It didn't look like the residual was used anywhere, so I just took it out. 
The residual here: 
https://github.com/JuliaRobotics/IncrementalInference.jl/blob/583a1ca2c468775e2e36cb7eb079f17bb8e8ddaa/src/NumericalCalculations.jl#L40
and here:
https://github.com/JuliaRobotics/IncrementalInference.jl/blob/583a1ca2c468775e2e36cb7eb079f17bb8e8ddaa/src/NumericalCalculations.jl#L54
is not populated or used

Reply:
I think you might not be looking at the code carefully enough.  To ensure the higher up calls are identical, each Roots and Minimize case has one redundant definition of the `residual`.  Thats why in some cases the parameter `residual` might look excessive.

-------

Look more closely at the Minimize snippet you posted.  The `residual` parameter **is being used** on line 61
https://github.com/JuliaRobotics/IncrementalInference.jl/blob/583a1ca2c468775e2e36cb7eb079f17bb8e8ddaa/src/NumericalCalculations.jl#L54-L61

as I tried to show here: https://github.com/JuliaRobotics/IncrementalInference.jl/pull/1147#discussion_r571094577

Is it necessary to drop the `::AbstractVector{<:Real}` type restriction for autodiff to work?

-----

In the Roots case, the lambda parameter `res` is a **requirement** from NLsolve.jl package, as I tried to show here:
https://github.com/JuliaRobotics/IncrementalInference.jl/pull/1147#discussion_r571093888

Reply:
perhaps we should add the comment:
```julia
# for code symmetry between Minimize (Optim.jl) and Roots (NLsolve.jl), we use standardized lambda `(res, x)-> obj...`
```

Reply:
i think the part I was missing earlier is you added residual in a new lambda at `nlsolve` instead.  I commented separately there a minute ago